wagtail/images/blocks.py

```diff
@@ -148,9 +148,16 @@ def to_python(self, value):
     def bulk_to_python(self, values):
         values = list(values)
 
-        if any(isinstance(value, int) for value in values):
-            # `values` is a list of image IDs (as we might encounter if an ImageChooserBlock has been
+        # The normal case: `values` is a list of dicts. This is the normal representation of this block.
+        if all(isinstance(value, dict) for value in values):
+            # `values` is a list of dicts containing `image`, `decorative` and `alt_text` keys
+            struct_values = super().bulk_to_python(values)
+
+        # Else we are in a fallback for backwards compatibility with ImageChooserBlock
+        else:
+            # `values` might be a list of image IDs (as we might encounter if an ImageChooserBlock has been
             # changed to an ImageBlock with no data migration)
+            # Or it could be a [None] if we are coming from an empty ImageChooserBlock
```

Comment:
Whether this comment is 100% technically accurate or not I am sure - it is what we are seeing switching some ImageChooserBlocks to ImageBlocks.